tutorials/interactive/index.md

```diff
@@ -0,0 +1,804 @@
+
+<img src="images/favorited-not-favorited.png" alt="the custom widget you'll create">
+
+To accomplish this, you'll create a single custom widget that includes both the
+star and and the count. Tapping the star changes state for both widgets,
+so the same widget should manage both.
+
+If you don't already have the starting code for the Lakes example,
+get it from the links in the blue [**Note**](#note) box.
+
+### Contents
+
+* [Stateful and stateless widgets](#stateful-stateless)
+* [Creating a stateful widget](#creating-stateful-widget)
+  * [Step 1: Decide who manages the widget's state](#step-1)
+  * [Step 2: Subclass StatefulWidget](#step-2)
+  * [Step 3: Subclass the State object](#step-3)
+  * [Step 4: Plug the stateful widget into the widget tree](#step-4)
+  * [Problems?](#problems)
+* [Managing state](#managing-state)
+  * [The widget manages its own state](#self-managed)
+  * [The parent manages the widget's state](#parent-managed)
+  * [A mix-and-match approach](#mix-and-match)
+* [Other interactive widgets](#other-interactive-widgets)
+  * [Standard widgets](#standard-widgets)
+  * [Material widgets](#material-widgets)
+* [Resources](#resources)
+
+<a name="stateful-stateless"></a>
+## Stateful and stateless widgets
+
+<div class="panel" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* Some layout widgets are stateful and some are stateless.
+* If a widget changes&mdash;the user interacts with it,
+  for example&mdash;it's _stateful_.
+* A widget's state consists of values that may change, like a slider's
+  current value, or whether a checkbox is checked.
+* A widget's state is stored in a State object, separating the widget's
+  state from its appearance.
+* When the widget's state changes, the state object calls
+  `setState`, telling the framework to redraw the widget.
+
+</div>
+
+Some layout widgets are stateful and some are stateless.
+
+A _stateless_ widget has no internal state to manage.
+[Icon,](https://docs.flutter.io/flutter/material/Icon-class.html)
+[IconButton,](https://docs.flutter.io/flutter/material/IconButton-class.html)
+and [Text](https://docs.flutter.io/flutter/widgets/Text-class.html) are
+examples of stateless widgets, which subclass
+[StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html).
+
+A _stateful_ widget is dynamic. The user can interact with a stateful widget
+(by moving a slider, or checking a box, for example),
+or it changes over time (perhaps a data feed causes the UI to update).
+[Checkbox,](https://docs.flutter.io/flutter/material/Checkbox-class.html)
+[Radio,](https://docs.flutter.io/flutter/material/Radio-class.html) and
+[Slider](https://docs.flutter.io/flutter/material/Slider-class.html) are
+examples of stateful widgets, which subclass
+[StatefulWidget](https://docs.flutter.io/flutter/widgets/StatefulWidget-class.html).
```

Comment:
i think it's misleading to use these as examples of stateful widgets, since they are only stateful because they're animated, not because they are interactive.